src/renderer/components/misc/MarkDown.vue

```diff
@@ -20,14 +23,29 @@ marked.use({
     },
 });
 
+const allowedTags = ["p", "em", "b", "strong", "ul", "ol", "li", "code", "pre", "blockquote", "span", "del", "body"];
+
+const allowedAttributes: string[] = [];
+
+if (props.allowLinks) {
+    allowedTags.push("a");
+
+    allowedAttributes.push("href");
+    allowedAttributes.push("target");
+
+    markdownRenderer.link = (href, title, text) => {
+        return `<a href="${href}" target="_blank">${text}</a>`;
```

Comment:
Check for XSS please!!! Both directly in the text, the href and with the javascript protocol

Reply:
https://brutelogic.com.br/blog/alternative-javascript-pseudo-protocol/

Reply:
more on XSS in general https://owasp.org/www-community/attacks/xss/

Reply:
Sadly a correct filter for it is not easy

Reply:
That part of the code just controls how the link tag is rendered. I added so that "target=_blank" attribute is added. The html generated by marked is later put through DOMPurify.sanitize. Also: allowLinks should not be exposed to end-user submitted entry. This pull does not enable link entry for DMs or lobby chat or anywhere else that end-users can submit markdown.